mathics/eval/strings.py

```diff
@@ -20,6 +21,27 @@
     "*": "*?",
 }
 
+if sys.version_info < (3, 11):
+    REGEX_UNICODE_PREFIX = r"(?u)"
+else:
+    REGEX_UNICODE_PREFIX = ""
+
+
+_SYMBOL_TO_REGEX_DICT = {
```

Comment:
Regarding if the dict is regenerated, is it still true if the items depends on a module variable?

Reply:
Indeed, the conditional is not necessary. It seems `(?u)` was needed for the `2.*` versions.

Reply:
> And I answered because you were curious and to verify and substantiate what I wrote.

BTW, thank you for that.

> 
> My concern is that you are thinking about this kind of stuff and we spend time at these kinds of things rather than focusing on other areas include these areas that _could_ save us a significant amount of run time. In the past the code has been perverted a bit chasing down such minor things.

The aim was not to obtain a "micro speed-up", but to avoid a "micro slow-down" if the conditional was needed. 

> 
> A little knowledge is dangerous.

Reply:
@rocky, in any case, I also had the feeling that putting that dictionary as module constant would be clearer than buried inside the routine, despite the possible overhead of using a module variable instead of a function variable. But again, I would prefer to leave that kind of decision to you.